src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useCallback } from 'react';
+import { FormikProps } from 'formik';
+import * as yup from 'yup';
+import { useHistory } from 'react-router-dom';
+import { ROOT_DOMAIN_ID } from '@colony/colony-js';
+
+import Dialog, { DialogProps } from '~core/Dialog';
+import { ActionForm } from '~core/Fields';
+
+import { Colony } from '~data/index';
+import { ActionTypes } from '~redux/index';
+import { WizardDialogType } from '~utils/hooks';
+import { pipe, withMeta, mapPayload } from '~utils/actions';
+
+import DialogForm from './EditDomainDialogForm';
+import { Color } from '~core/ColorTag';
+
+export interface FormValues {
+  domainId: string;
+  domainName: string;
+  domainColor?: Color;
+  domainPurpose?: string;
+  annotationMessage?: string;
+}
+
+interface CustomWizardDialogProps {
+  prevStep?: string;
+  colony: Colony;
+  selectedDomainId?: string;
+}
+
+type Props = DialogProps &
+  Partial<WizardDialogType<object>> &
+  CustomWizardDialogProps;
+
+const displayName = 'dashboard.EditDomainDialog';
+
+const EditDomainDialog = ({
+  callStep,
+  prevStep,
+  cancel,
+  close,
+  colony,
+  colony: { colonyAddress, colonyName, domains },
+  selectedDomainId,
+}: Props) => {
+  const history = useHistory();
+
+  const validationSchema = yup.object().shape({
+    domainName: yup.string(),
+    domainId: yup.number().required(),
+    domainColor: yup.string(),
+    domainPurpose: yup.string(),
+    annotationMessage: yup.string().max(4000),
+  });
```

Comment:
So that will be `20` and `90`: https://github.com/JoinColony/colonyDapp/blob/5ce5c8d735f69f5b1caf7851ee4ebfa05fa9976e/src/modules/dashboard/components/EditDomainDialog/EditDomainDialogForm.tsx#L156-L170

So something like this:
```suggestion
  const validationSchema = yup.object().shape({
    domainName: yup.string().max(20),
    domainId: yup.number().required(),
    domainColor: yup.string(),
    domainPurpose: yup.string().max(90),
    annotationMessage: yup.string().max(4000),
  });
```

Yes, I know that could be circumvented in other places, but that doesn't mean we can't properly validate it :)

PS: please add the same validation to the create domain modal